google/generativeai/files.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import os
+import pathlib
+import mimetypes
+from typing import Iterable
+
+from google.generativeai.types import file_types
+
+from google.generativeai.client import get_default_file_client
+
+__all__ = ["File", "create_file", "get_file", "list_files"]
+
+
+def create_file(
+    path: str | pathlib.Path | os.PathLike,
+    *,
+    mime_type: str = None,
+    name: str | None = None,
+    display_name: str = None,
+) -> file_types.File:
+    client = get_default_file_client()
+
+    path = pathlib.Path(os.fspath(path))
+
+    if mime_type is None:
+        mime_type, _ = mimetypes.guess_type(path)
+
+    if name is not None and "/" not in name:
+        name = f"files/{name}"
+
+    if display_name is None:
+        display_name = path.name
+        print(display_name)
+
+    response = client.create_file(
```

Comment:
this should be upload_file? I'm also getting an error: `TypeError: FileServiceClient.create_file() got an unexpected keyword argument 'path'`

Reply:
Yes, fixed.